sw/tools/generators/gen_settings.ml

```diff
@@ -90,13 +90,16 @@ let print_dl_settings = fun out settings settings_xml ->
         try
           let h = ExtXml.attrib s "handler" in
           begin
+            let t = ExtXml.attrib_or_default s "type" "" in
```

Comment:
I a got it right, you want to call a function without a variable associated. If yes, this is something I wanted to do for a long time, however this implementation might be limited. It is still necessary to associate a variable. Also, the `type` attribute seems related to persistent settings, is it breaking this "feature" (not sure if it is really working, but still). Also, I wanted to make an "automatic" variable to get the return value of the function, what do you think ?

Reply:
Yes you got it totally correct. This method actually works perfectly fine and no variable needs to be created, the `var` tag is still needed for naming and other purposes. But in the generated code it isn't used at all anymore because of the `type="fun"`.
The type "fun" doesn't work together with setting it persistent as well, but that should generate an error by `gen_settings`. The other persistent settings just work fine.

For the option to get the return value of a function it could be added really easily as well, currently it generates something like this:
```
#define DlSetting(_idx, _value) { \
  switch (_idx) { \
...
    case 10: mcu_reboot( _value ); break;\
...
}

#define PeriodicSendDlValue(_trans, _dev) { \
  static uint8_t i;\
  float var;\
  if (i >= 88) i = 0;\
  switch (i) { \
...
case 10: var = 0; break;\
...
}

static inline float settings_get_value(uint8_t i) {
  switch (i) {
...
    case 10: return 0;
...
}
```
Which could of course also be something like this:
```
case 10: _value = mcu_reboot( _value ); break;\
...
case 10: var = some_other_func(); break;\
...
case 10: return some_other_func();
```

I'm not that great at ocaml and the generators etc. so if you have better suggestions that also work, please let me know. This was just an easy fix for something that was annoying me for years.

See for example: https://github.com/paparazzi/paparazzi/pull/2859/files/9419301763072d0c76a3b8e42dae0c2509cb7bc2#diff-b0b28a5c9934bc204875c10ce4060dd75bcc3b0956e89142f3cd8e5bd878a1afR10